syntax = "proto3"; //声明了protobuf的版本

package fixbug;//声明了代码所在的包 (对于C++来说是namespace)

// 定义下面的选项 表示生成service服务类和rpc方法描述 默认不生成
option cc_generic_services = true;
message ResultCode
{
    int32 errcode = 1;
    bytes errmsg = 2;
}
// 定义登录请求消息类型 name pwd
// 这里的string类型 是protobuf自己的类型
// 这里定义成bytes类型 提升效率
message LoginRequest
{
    bytes name = 1;
    bytes pwd = 2;
    //map<int32, string> test = 3;
}

//定义登录相应消息类型
message LoginResponse
{
    ResultCode result = 1;
    bool success = 2;
}

message RegisterRequest
{
    uint32 id = 1;
    bytes name = 2;
    bytes pwd = 3;
}

message RegisterResponse
{
    ResultCode result = 1;
    bool success = 2;
}

// 在protobuf里面定义描述rpc方法的类型 -service
// protobuf不支持rpc 只是对rpc方法的描述
// 无论是message 还是service 在生成C++代码后都是i要给class类
// 只写下面的service 在生成C++代码后默认不会生成类的  需要在问价最前面加上选项
service UserServiceRpc
{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc Register(RegisterRequest) returns(RegisterResponse);
}